src/pykiso/lib/auxiliaries/udsaux/uds_auxiliary.py

```diff
@@ -250,3 +283,38 @@ def _abort_command(self) -> None:
     def _run_command(self, cmd_message, cmd_data=None) -> Union[dict, bytes, bool]:
         """Not used."""
         pass
+
+    def _get_instance(self):
+        return self
+
+
+class TesterPresentSender(threading.Thread):
+    """Self that continuously sends tester present messages via UDS"""
+
+    TESTER_PRESENT_SEND = b"\x3E\x00"
+
+    def __init__(self, uds_aux: UdsAuxiliary, period: float = 4):
+        """Constructor
+
+        :param period: period to use for the cyclic sending of tester present
+        :param uds_aux: auxiliary used to send tester present
+        """
+        self.uds_aux = uds_aux
+        self.period = period
+        self.stop_event = threading.Event()
+        super().__init__(name="TesterPresentSender")
+
+    def run(self):
+        """send tester present at defined period until stopped"""
+        while not self.stop_event.is_set():
+            self.uds_aux.transmit(self.TESTER_PRESENT_SEND, self.uds_aux.req_id)
```

Comment:
You should use uds_aux.send_uds_raw instead as transmit just sends the raw bytes and doesn't go through the ISO TP layer to encode them before sending

Reply:
Done